### ISSUE TITLE
CRT support for signing?

### ISSUE THREAD

**Issue by tomato42** [hacl-star/hacl-star#884]
**Is your feature request related to a problem? Please describe.**
The implementation of RSAPSS doesn't seem to be using the CRT optimisation: 
https://github.com/hacl-star/hacl-star/blob/d59b0b08bc07e8f65490c9cfb12d1c3f51f48f61/code/rsapss/Hacl.Spec.RSAPSS.fst#L317-L324

**Describe the solution you'd like**
It would be nice if the code used the Chinese Reminder Theorem optimisation, as that would increase performance by a factor of 2 to 3

**Describe alternatives you've considered**
Not using CRT hurts performance in typical case

**Additional context**
Basically every production grade implementation of RSA uses the CRT approach for the RSA private key operation.

That being said, I'm far from familiar with F*, so I may be reading the code completely incorrectly, feel free to point out to me that I'm reading the code incorrectly or looking at wrong piece of code.

**Comment by karthikbhargavan** [comment]
Good point, we do not do this yet, because the code is designed to be super general.

**Comment by tomato42** [comment]
OK, but
1. it's a significant performance difference
2. 99% of keys people will want to use in practice will have CRT coefficients included
3. it's possible to recover _p_, and _q_ from _d_ using algorithm specified in appendix C (page 118) of NIST [SP-800-56B rev 1](https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-56Br1.pdf)
4. the code can use either algorithm, depending on which coefficients were provided
